### PR TITLE
Updated docs

### PR DESCRIPTION
**Pull request by purnatag**
Closes #14.  Updated docs from the text [here](https://www.notion.so/subcom/Overview-SimpleAccumulator-2abff9b8aa1d4a3e9e608aa37ab9add7?d=b54207d1d7f647099b3ddf063f642010).

`examples/error_mean.rs` is now a module inside `src/lib.rs` and the code in said module is inactive due to cfg directives.

The reason for above is that `mean` used in the `error_mean` module is renamed `buffer_mean` and made private.
Similarly, `variance` is renamed `buffer_variance` and made private.
`r_mean` is now `mean`.
`r_variance` is now `variance`.
`r_total` is now `total`.

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -15,6 +15,7 @@ documentation = "https://docs.rs/simple_accumulator"
 
 [dependencies]
 num = "0.4.0"
+plotly = "0.8.3"
```
Comment (dilawar) [inline Cargo.toml:18]: This should not be a dependency. It can be a dev- or test- dependency.

File: src/lib.rs [hunk 2]
```diff
@@ -462,30 +462,37 @@ fn median_select(data: &Vec<f64>, k: usize) -> Option<f64> {
     }
 }
 
-
 /*============================================ MAIN OPERATIONS : push, append, remove, pop ============================================= */
 impl SimpleAccumulator {
-    /// Similar to `push` in vector `Vec`. When `fixed_capacity` flag is 'true' and ring buffer is full
-    /// rewrites the oldest element with the latest, following FIFO order. 
+    /// Function similar to `push` in vector `Vec`. When `fixed_capacity` is 'true' and ring buffer is full,
```
Comment (dilawar) [inline src/lib.rs:467]: `default` should not set `fixed_capacity=true`.

File: src/lib.rs [hunk 3]
```diff
@@ -462,30 +462,37 @@ fn median_select(data: &Vec<f64>, k: usize) -> Option<f64> {
...
-        self.r_total += 1;
-        let r_delta = y - self.r_mean;
-        let r_delta_n = r_delta / self.r_total as f64;
-        self.r_mean += r_delta_n;
-        let r_term1 = r_delta * r_delta_n * (self.r_total as f64 - 1.0);
-        let r_stats1 = self.r_variance * (self.r_total as f64 - 2.0) + r_term1;
-        self.r_variance = r_stats1 / (self.r_total as f64 - 1.0);
+        self.total += 1;
+        let delta = y - self.mean;
+        let delta_n = delta / self.total as f64;
```
Comment (dilawar) [inline src/lib.rs:485]: This is wrong I think. `delta / self.total as f64` is `(delta / self.total) as f64` which is not `(delta as f64) / (self.total as f64)`.  Both of these are very different.

**Review by dilawar — CHANGES_REQUESTED** [review]

File: src/lib.rs (hunk 2 above)
Comment (sn99) [reply src/lib.rs:467]: I don't think we should change how `default` trait works for `bool` types in rust, instead, we can make something like `SimpleAccumulator::FixedDefault` and that can set `fixed_capacity=false` for it.

**Pull request merged**